extensions/ql-vscode/src/pure/result-keys.ts

```diff
@@ -1,33 +1,45 @@
 import * as sarif from 'sarif';
 
+/**
+ * Identifies a result, a path, or one of the nodes on a path.
+ */
+interface ResultKeyBase {
+  resultIndex: number;
+  pathIndex?: number;
+  pathNodeIndex?: number;
+}
+
 /**
  * Identifies one of the results in a result set by its index in the result list.
  */
-export interface Result {
+export interface Result extends ResultKeyBase {
   resultIndex: number;
+  pathIndex?: undefined;
+  pathNodeIndex?: undefined;
 }
 
 /**
  * Identifies one of the paths associated with a result.
  */
-export interface Path extends Result {
+export interface Path extends ResultKeyBase {
   pathIndex: number;
+  pathNodeIndex?: undefined;
 }
 
 /**
  * Identifies one of the nodes in a path.
  */
-export interface PathNode extends Path {
+export interface PathNode extends ResultKeyBase {
+  pathIndex: number;
   pathNodeIndex: number;
 }
 
-/** Alias for `undefined` but more readable in some cases */
-export const none: PathNode | undefined = undefined;
+export type ResultKey = Result | Path | PathNode;
 
 /**
  * Looks up a specific result in a result set.
  */
-export function getResult(sarif: sarif.Log, key: Result): sarif.Result | undefined {
+export function getResult(sarif: sarif.Log, key: Result | Path | PathNode): sarif.Result | undefined {
   if (sarif.runs.length === 0) return undefined;
```

Comment:
Minor: The body of this function can be replaced with this:

```
  return sarif.runs?.[0]?.results?.[key.resultIndex];
```

It's really outside the scope of this PR, but it's a minor readability improvement if you want to add it.